sinonjs/sinon
# Add subdir eslintrc for mjs

#### Purpose
While https://github.com/sinonjs/sinon/pull/1715 added `.mjs` files and a nifty `lint-mjs` command, it leaves the issue of the configuration being done on the command line instead of in a config file.  The reason it's an issue is that eslint-integrated editors (like VSCode, which I'm using) will complain about the use of `import` in the `.mjs` files, for example.

This modification simply adds a subdirectory `.eslintrc.yml` file in the mjs test directory which specifies the proper eslint option for those files.  Eslint-aware editors should stop complaining.

This also allowed me to simplify the `lint-js` command to incorporate what `lint-mjs` was doing.  Now `npm run lint` only spawns two processes instead of three.

I also added all `.yml` and `.json` files to the `.editorconfig` file.  It was only looking at `package.json` and `.travis.yml`, causing files like the new `.eslintrc.yml` file to have the wrong number of spaces for indentation in my editor.

#### How to verify - mandatory
1. Check out this branch
2. `npm install`
3. `npm run lint` or `npm run lint-js`

#### Checklist for author
